test/lib/sdkdriver/src/transport/websocket/server.ts

```diff
@@ -0,0 +1,85 @@
+import Transport from '../';
+import { EventEmitter } from 'events';
+import * as WebSocket from 'ws';
+
+/**
+ * WebSocket server {@link Transport}.
+ * @class
+ * @fires Transport#close
+ * @fires Transport#message
+ */
+export default class WSServerTransport extends EventEmitter implements Transport {
+  private readonly  _deps: any;
+  private readonly _sendBuffer: Array<string>;
+  private readonly _wsServer: WebSocket.Server;
+  private _wsConnection: WebSocket | null;
+
+  /**
+   * Constructor.
+   * @param {number} wsPort - The WebSocket server port.
+   * @param {TestDriver} testDriver - The {@link TestDriver} that connects
+   *   to the {@link WSServerTransport}
+   */
+  constructor(wsPort: number, deps: any = {}) {
+    super();
+    this._deps = {
+      WebSocket,
+      ...deps
+    };
+    this._sendBuffer = [];
+    this._wsConnection = null;
+    this._wsServer = new this._deps.WebSocket.Server({ port: wsPort });
+  }
+
+  /**
+   * Close the {@link WSServerTransport}.
+   * @returns {void}
+   */
+  close(): void {
+    if (this._wsConnection) {
+      try {
+        this._wsConnection.close();
+      } catch (e) {
+        // Do nothing.
+      }
+      this._wsConnection = null;
+    }
+
+    try {
+      this._wsServer.close();
```

Comment:
Should we gate on `if (this._wsServer)` here?

Reply:
@markandrus `._wsServer` is never null.